src/HSA/libhsa_api.jl

```diff
@@ -481,8 +481,8 @@ function memory_free(ptr::Ptr{T}) where T
     ccall((:hsa_memory_free, "libhsa-runtime64"), Status, (Ptr{T},), ptr)
 end
 
-function memory_copy(dst::Ref{Cvoid}, src::Ref{Cvoid}, size::Integer)
-    ccall((:hsa_memory_copy, "libhsa-runtime64"), Status, (Ref{Cvoid}, Ref{Cvoid}, Csize_t), dst, src, size)
+function memory_copy(dst::Ptr{T}, src::Ptr{T}, size::Integer) where T
```

Comment:
If we need to use `Ptr` to please the HSA API, okay. I had the generator default to `Ref` in most cases as a safer default when making these originally. One advantage you get from using `Ref` is that you can (as needed) relax the method arguments, and allow an array in--Julia will autoconvert a generic array to Ref, with the correct offset, when you provide one to a `ccall` expecting `Ref`. Otherwise, you'll need to do pointer conversions manually before passing it in.

Reply:
...also I seem to remember `ccall` getting angry when you have runtime-dependent arguments

Reply:
Thanks for the feedback!

Well, this function is called with pointers that can be in any address space, including device memory, and are not necessarily managed by Julia's GC, therefore I decided to go with `Ptr`.
Also, it is only called by three functions in the `Mem` module that call `GC.@preserve` as needed.
I understand the perks of using `Ref` instead of `Ptr`, but it seems simpler to just use pointers in this case.

Reply:
That's a good point, we'll need to use this to transfer memory between device pointers which don't lend themselves well to being boxed up in a `Ref`. So I'm ok with this isolated instance.

However, for other API calls, I like the benefits we get from `Ref`-typing, as @wsphillips indicated (also, TIL!), so let's try to keep those if that's sensible.

Reply:
Sure--I trust that if you're down there working with it, there's reason for it.